src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py

```diff
@@ -127,9 +127,12 @@ def isDrainMode(config):
 
     reqMgrAux = ReqMgrAux(config.TaskArchiver.ReqMgr2ServiceURL)
     agentConfig = reqMgrAux.getWMAgentConfig(config.Agent.hostName)
-
-    return agentConfig["UserDrainMode"] or agentConfig["AgentDrainMode"]
-
+    if "UserDrainMode" in agentConfig and "AgentDrainMode" in agentConfig:
+        AUXDB_AGENT_CONFIG_CACHE = agentConfig
```

Comment:
Isn't it another variable/id? I think we'd need to declare it `global` if we want to overwrite its value in here, no?

Reply:
Yes you are right. Thanks I will update that.